modules/acknowledgements/jsx/acknowledgementsIndex.js

```diff
@@ -0,0 +1,370 @@
+    };
+
+    this.fetchData = this.fetchData.bind(this);
+    this.pickElements = this.pickElements.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+    this.openModalForm = this.openModalForm.bind(this);
+    this.closeModalForm = this.closeModalForm.bind(this);
+    this.renderCitationPolicy = this.renderCitationPolicy.bind(this);
+    this.renderAddForm = this.renderAddForm.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   * Additionally add hiddenHeaders to global loris variable
+   * for easy access by columnFormatter.
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  pickElements(formElement, keys) {
+    let subset = {};
+    keys.forEach((key) => {
+      if (formElement.hasOwnProperty(key)) {
+        subset[key] = formElement[key];
+      }
+    });
+    return subset;
+  }
+
+  /**
+   * Store the value of the element in this.state.formData
+   *
+   * @param {string} formElement - name of the form element
+   * @param {string} value - value of the form element
+   */
+  setFormData(formElement, value) {
+    const formData = this.state.formData;
+    formData[formElement] = value;
+    this.setState({
+      formData: formData,
+    });
+  }
+
+  /**
+   * Handles the submission of the Add Acknowledgements form
+   *
+   * @param {event} e - event of the form
+   */
+  handleSubmit(e) {
+    let formData = this.state.formData;
+    let formObject = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObject.append(key, formData[key]);
+      }
+    }
+    formObject.append('fire_away', 'Add');
+    $.ajax({
+      type: 'POST',
+      url: loris.BaseURL + '/acknowledgements/',
+      data: formObject,
+      cache: false,
+      contentType: false,
+      processData: false,
+      success: (data) => {
+        swal('Success!', 'Acknowledgement added.', 'success');
+        this.fetchData();
+      },
+      error: (error) => {
+        console.error(error);
+        let message = error.responseText;
+        swal('Error!', message, 'error');
+      },
+    });
+  }
+
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
+    return <td>{cell}</td>;
```

Comment:
Should we always wrap or leave as is?
```
return (
<td>{cell}</td>
);
```

Reply:
I personally prefer inline formatting when possible.

Reply:
@HenriRabalais Okay, we can save two lines. This reminds me of me not using curly braces when only a single line of code after a for loop or an if statement. Took me awhile to break that habit.